queries/robot/highlights.scm

```diff
@@ -77,3 +81,50 @@
 (try_statement
   (else_statement
     "ELSE" @keyword.exception))
+
+; Highlight text chunks in "Documentation" settings with @spell
+
+(setting_statement
+  name: (setting_name) @_name
+  (arguments
+    (argument
+      (text_chunk) @spell))
+  (#match? @_name "^Documentation$"))
```

Comment:
For performance reasons, `#match?` should not be used in our queries. You should replace this with `#eq?`.

Reply:
And `@spell` does not highlight anything. Don't you want to add `@string.documentation` as well for actual highlights?

Reply:
Setting names in Robot are case insensitive, so I was hoping there was a way to match "documentation" case insensitively with `match`. Is that possible?

I know `@spell` doesn't highlight anything, but as far as I understood, it should still be put in the highlight query file (?). The issue I'm addressing is marking documentation keywords for spell checking. I didn't know about `@string.documentation`, it makes sense to add that too.

Reply:
> Is that possible?

Yes, but not like this, and then there are still better ways. do you expect you need to support the full `DOcumenTATION` combinatorial explosion? If not, use `#any-of?` for the reasonable variants; otherwise use `#lua-match?` with a `[dD][oO]...` Lua pattern.

Reply:
> I know @spell doesn't highlight anything, but as far as I understood, it should still be put in the highlight query file (?). The issue I'm addressing is marking documentation keywords for spell checking. 

Yes, that is correct. I was just replying to the comment, which explicitly mentions "highlighting".

Reply:
Pro-tip: always back-tick captures in commit messages; otherwise the poor user named spell will get pinged on any force-push.

Reply:
> > Is that possible?
> 
> Yes, but not like this, and then there are still better ways. do you expect you need to support the full `DOcumenTATION` combinatorial explosion? If not, use `#any-of?` for the reasonable variants; otherwise use `#lua-match?` with a `[dD][oO]...` Lua pattern.

I suppose it's not really reasonable for anybody to write Documentation like that, so I'm settling for `#any-of?`.

Reply:
Can always revisit if someone comes up with a reasonable request. The main point is to avoid `#match?` like the plague (because it spins up Vim's own regex engine) and to prefer literal matches over patterns wherever possible.